src/cps/search.cpp

```diff
@@ -71,36 +71,71 @@ namespace cps::search {
             return out;
         }
 
+        fs::path libdir() {
+            // TODO: libdir needs to be configurable based on the personality,
+            //       and different name schemes.
+            //       This is complicated by the fact that different distros have
+            //       different schemes.
+            return "lib";
```

Comment:
Seems sensible.

Should the distro have a configuration file or should there be environment variables for these options? I'm not sure we want to have to patch `cps-config` for every possible distro.

Reply:
Yes, my follow up work for a rebased #31 would use meson's `get_option('libdir')` and CMake's `CMAKE_INSTALL_LIBDIR` to populate both this and the datadir, I just didn't include that here because I've been trying to keep the PRs smaller and more manageable.

Reply:
I went ahead and added the patches to set the libdir and datadir from the build system. They're not too invasive, and I think their suitable to land as-is.

Reply:
Well never mind, it's breaking the pixi build and I would rather handle that separately. I'll open a second PR for that and we can work out how to fix pixi